### PR TITLE
backport: add regexes support

### PR DESCRIPTION
**Pull request by sileht**
Closes: #494

### CODE REVIEW

File: mergify_engine/actions/backport.py [hunk 1]
```diff
@@ -28,17 +29,40 @@
 LOG = daiquiri.getLogger(__name__)
 
 
+def regex(value):
```
Comment (jd) [inline mergify_engine/actions/backport.py]: I tend to prefer those voluptuous def with a capital e.g. `Regex`

File: mergify_engine/actions/backport.py [hunk 2]
```diff
@@ -28,17 +29,40 @@
...
 
     def run(self, installation_id, installation_token,
             event_type, data, pull, missing_conditions):
         if not pull.g_pull.merged:
             return None, "Waiting for the pull request to get merged", ""
 
+        branches = self.config['branches']
+        if self.config['regexes']:
+            regexes = list(map(re.compile, self.config['regexes']))
+            print(regexes)
```
Comment (jd) [inline mergify_engine/actions/backport.py]: nop

File: mergify_engine/actions/backport.py [hunk 3]
```diff
@@ -28,17 +29,40 @@
...
+            branches.extend((
+                branch.name
+                for branch in pull.g_pull.base.repo.get_branches()
+                if any(map(lambda regex: regex.match(branch.name), regexes))
+            ))
+        print(pull.g_pull.base.repo.get_branches())
+        print(branches)
```
Comment (jd) [inline mergify_engine/actions/backport.py]: nop

File: mergify_engine/actions/backport.py [hunk 4]
```diff
@@ -28,17 +29,40 @@
...
+        branches = self.config['branches']
+        if self.config['regexes']:
+            regexes = list(map(re.compile, self.config['regexes']))
+            print(regexes)
+            branches.extend((
+                branch.name
+                for branch in pull.g_pull.base.repo.get_branches()
+                if any(map(lambda regex: regex.match(branch.name), regexes))
+            ))
+        print(pull.g_pull.base.repo.get_branches())
```
Comment (jd) [inline mergify_engine/actions/backport.py]: nop

File: mergify_engine/actions/backport.py [hunk 5]
```diff
@@ -28,17 +29,40 @@
...
+    except re.error as e:
+        raise voluptuous.Invalid(str(e))
+    return value
+
+
 class BackportAction(actions.Action):
-    validator = {voluptuous.Required("branches", default=[]): [str]}
+    validator = {
+        voluptuous.Required("branches", default=[]): [str],
+        voluptuous.Required("regexes", default=[]): [regex]
```
Comment (jd) [inline mergify_engine/actions/backport.py]: Not a fan of the `regexes` name but I can't think of anything better.

Comment (sileht) [reply mergify_engine/actions/backport.py]: me neither :(

Comment (adamjstone) [reply mergify_engine/actions/backport.py]: "expressions"? Too late, I know.

**Comment by adamjstone** [comment]
Thanks, gentlemen! :smile:

**Pull request merged**